core/GlobalState.cc

```diff
@@ -584,6 +584,17 @@ constexpr decltype(GlobalState::STRINGS_PAGE_SIZE) GlobalState::STRINGS_PAGE_SIZ
 // look up a symbol whose flags match the desired flags. This might look through mangled names to discover one whose
 // flags match. If no sych symbol exists, then it will return noSymbol.
 SymbolRef GlobalState::lookupSymbolWithFlags(SymbolRef owner, NameRef name, u4 flags) const {
+    return lookupSymbolSuchThat(owner, name, [&](SymbolRef res) { return (res.data(*this)->flags & flags) == flags; });
+}
+
+SymbolRef GlobalState::lookupMethodSuchThat(SymbolRef owner, NameRef name, function<bool(SymbolRef)> pred) const {
```

Comment:
function has a huge overhead(in particular when you pass it by copy).
This method is also only used once.
Can we not introduce this abstraction but instead write more specific versions (possibly duplicating parts of implementation)?